### PR TITLE
feat(sbt): Update scala version

### LINKED ISSUE #6785: [sbt] Suggest Scala version upgrades
**What would you like Renovate to be able to do?**

The high level goal is to be notified of newer Scala versions.

With maven it usually works out of the box because Scala is used as a regular Maven dependency.

With sbt, the Scala version is defined as a first level attribute of the `build.sbt` file:
```scala
scalaVersion := "2.12.11"
```

And for now, this version isn't picked up by RenovateBot to check for newer versions.

_Additional details_

- scala versions availability can be checked against the regular `org.scala-lang:scala-library` maven dependency (https://mvnrepository.com/artifact/org.scala-lang/scala-library)
- what about dependencies? Scala dependencies are dependent against the scala version (minor) but I _don't_ expect Renovate to check that all dependencies are available for a Scala version before suggesting an upgrade. The CI pipeline should raise that issue immediately saying that a dependency is not available.
- sbt allows "cross building" which means that the project is actually compiled multiple times: one against each scala version declared as the following: `crossScalaVersions := Seq("2.12.11", "2.13.2")`. This is optional though, not every project use it but a lot of libraries do. In such case, I would expect:
  - this line to be updated to be consistent with the `scalaVersion` property
  - this line to only be updated with bugfix versions: it would not make sense to suggest to update `crossScalaVersions := Seq("2.13.2", "2.13.2")` but `crossScalaVersions := Seq("2.12.12", "2.13.2")` would. When saying that, I realize this could be rather tricky.. this is maybe a "bonus" feature and focusing on the simple no-cross-build projects can be a first step.

**Did you already have any implementation ideas?**

N/A

**Are there any workarounds or alternative ideas you've tried to avoid needing this feature?**

Scala versions upgrades are not so frequent so we are usually notified via Twitter or someone around us that a new versions is available.

**Is this a feature you'd be interested in implementing yourself?**

I'll he bappy to if someone can provide some highlevel view of what would actually need to be done in Renovate terms (which files to update or look into...).

### PR DESCRIPTION
**Pull request by zharinov**
Closes #6785

### CODE REVIEW

**Comment by zharinov** [comment]
Example repo: https://github.com/zharinov/renovate-test-sbt/pulls

File: lib/manager/common.ts [hunk 1]
```diff
@@ -172,6 +172,9 @@ export interface PackageDependency<T = Record<string, any>> extends Package<T> {
   autoReplaceStringTemplate?: string;
   depIndex?: number;
   editFile?: string;
+  separateMinorPatch?: boolean;
+  separateMajorMinor?: boolean;
+  separateMultipleMajor?: boolean;
```
Comment (zharinov) [inline lib/manager/common.ts:177]: Actually, I don't think it's better way to make these fields availabe as `extract.ts` results, but the code already can deal with it.

Comment (rarkins) [reply lib/manager/common.ts:177]: Do we need them all now, or just minorpatch?

File: lib/manager/sbt/extract.ts [hunk 2]
```diff
@@ -207,7 +208,14 @@ function parseSbtLine(
   if (!isComment(line)) {
     if (isScalaVersion(line)) {
       isMultiDeps = false;
-      scalaVersion = normalizeScalaVersion(getScalaVersion(line));
+      const rawScalaVersion = getScalaVersion(line);
+      scalaVersion = normalizeScalaVersion(rawScalaVersion);
+      dep = {
+        datasource: datasourceMaven.id,
+        depName: 'org.scala-lang:scala-library',
```
Comment (rarkins) [inline lib/manager/sbt/extract.ts:215]: What about using depName=scala and lookupName=org.scala-lang:scala-library ?

File: lib/manager/common.ts (hunk 1 above)
Comment (zharinov) [reply lib/manager/common.ts:177]: I'm thinking about extracting these three options into separate interface, and them mix it into both `PackageDependency` and `LookupUpdateConfig`

**Review by viceice — APPROVED** [review]

**Pull request merged**